test/test_demos.py

```diff
@@ -0,0 +1,41 @@
+"""
+Checks that all demo scripts run.
+"""
+import glob
+import os
+import pytest
+import shutil
+import subprocess
+import sys
+
+
+# Set environment variable to specify that tests are being run so that we can
+# cut down the length of the demos
+os.environ["ANIMATE_REGRESSION_TEST"] = "1"
+
+cwd = os.path.abspath(os.path.dirname(__file__))
+demo_dir = os.path.abspath(os.path.join(cwd, "..", "demos"))
+all_demos = glob.glob(os.path.join(demo_dir, "*.py"))
+
+
+@pytest.fixture(params=all_demos, ids=lambda x: os.path.basename(x))
+def demo_file(request):
+    return os.path.abspath(request.param)
+
+
+def test_demos(demo_file, tmpdir, monkeypatch):
+    assert os.path.isfile(demo_file), f"Demo file '{demo_file}' not found."
+    demo_name = os.path.splitext(os.path.basename(demo_file))[0]
+
+    # Copy mesh files
+    source = os.path.dirname(demo_file)
+    for f in glob.glob(os.path.join(source, "*.msh")):
+        shutil.copy(f, str(tmpdir))
+
+    # Change working directory to temporary directory
+    monkeypatch.chdir(tmpdir)
+    subprocess.check_call([sys.executable, demo_file])
+
+    # Clean up plots
+    for ext in ("jpg", "pvd", "vtu"):
+        subprocess.check_call(["rm", "-f", f"{demo_name}*.{ext}"])
```

Comment:
So you're assuming a convention where all generated output files are named <basename>.jpg/etc? Sounds sensible but is already not true for the first demo - and isn't strictly followed in goalie either, e.g. burgers_oo.py has burgers-oo_ee.jpg
I'm generally a little wary of duplicating this sort of logic here, that's typically also in the Makefile.

Reply:
Ah good spot. Fixed in https://github.com/pyroteus/animate/pull/63/commits/6137960d692703273d6eb7295c7421455013915d.